include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -346,6 +447,23 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUBuffer** const _begin
     }
     finalizeBlock();
 
+    cmdbuf->end();
+    IGPUQueue::SSubmitInfo submit;
+    submit.commandBufferCount = 1u;
+    auto* cmdbuf_ptr = cmdbuf.get();
+    submit.commandBuffers = &cmdbuf_ptr;
+    // TODO have one signal semaphore here and write it into SParams
+    submit.signalSemaphoreCount = 0u;
+    submit.pSignalSemaphores = nullptr;
+    submit.waitSemaphoreCount = 0u;
+    submit.pWaitDstStageMask = nullptr;
+    submit.pWaitSemaphores = nullptr;
+    // dont event tell the queue to signal a fence after last submit if it'll never be touched by user
+    auto* signalFence = _params.transferFence ? fence.get() : nullptr;
+    _params.transferQueue->submit(1u, nullptr, signalFence);
+    if (_params.transferFence)
+        _params.transferFence[0] = std::move(fence);
+
```

Comment:
This is the reason for my other comment 
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28/files/236b8ecdf029cd54251c4f120fd98300c23b6fb7..91b055e35d412c474cc3c8bec8802ffd2681bd9d#r614272961

you need a lot of flexibility here, because you might have to make the compute queue wait for the transfer (i.e. buffer to image upload, then mip-map with compute synced with a semaphore, or with an event if its the same queue used for both)

so you should only worry about building the right command buffer in the `create` calls (with the possibility of "auto submission" if some sort of overflow happens, but such a submit would be without any fence or semaphore) and have the semaphored and fenced submission happen outside of the `create` (if such is needed at all).

The last/top level submit from `getGPUObjectsFromAssets` which submits the final commandbuffer can signal the actual fences, sempahores and events in the `IGPUObjectFromAssetConverter::SParams`